app/domain/authentication/authn_jwt/fetch_issuer_value.rb

```diff
@@ -0,0 +1,139 @@
+require 'uri'
+
+module Authentication
+  module AuthnJwt
+
+    # FetchIssuerValue command class is responsible to fetch the issuer secret value,
+    # in order to validate it later against the JWT token claim
+    FetchIssuerValue ||= CommandClass.new(
+      dependencies: {
+        resource_class: ::Resource,
+        fetch_secrets: ::Conjur::FetchRequiredSecrets.new,
+        logger: Rails.logger
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      extend(Forwardable)
+      def_delegators(:@authenticator_input, :service_id, :authenticator_name,
+                     :account)
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingIssuerConfigurationValue.new)
+        fetch_issuer_value
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchedIssuerValueFromConfiguration.new)
+
+        @issuer_value
+      end
+
+      private
+
+      # fetch_issuer_value function is responsible to fetch the issuer secret value,
+      # according to the following logic:
+      # Fetch from `issuer` authenticator resource,
+      # In case `issuer` authenticator resource not configured, than only 1 resource, `provider-uri` or `jwks-uri`,
+      # should be configured.
+      # So the priority is:
+      # 1. issuer
+      # 2. provider-uri or jwks-uri
+      # In case the resource is configured but the not initialized with secret, throw an error
+      def fetch_issuer_value
+        if issuer_resource_exists?
+          @logger.debug(LogMessages::Authentication::AuthnJwt::IssuerResourceNameConfiguration.new(resource_id(ISSUER_RESOURCE_NAME)))
+
+          @issuer_value=issuer_secret
+        else
+          validate_issuer_configuration
+
+          if provider_uri_resource_exists?
+            @logger.debug(LogMessages::Authentication::AuthnJwt::IssuerResourceNameConfiguration.new(resource_id(PROVIDER_URI_RESOURCE_NAME)))
+
+            @issuer_value=provider_uri_secret
+          elsif jwks_uri_resource_exists?
+            @logger.debug(LogMessages::Authentication::AuthnJwt::IssuerResourceNameConfiguration.new(resource_id(JWKS_URI_RESOURCE_NAME)))
+
+            @issuer_value=fetch_issuer_from_jwks_uri_secret
+          end
+        end
+
+        @logger.debug(LogMessages::Authentication::AuthnJwt::RetrievedIssuerValue.new(@issuer_value))
+      end
+
+      def issuer_resource_exists?
+        !issuer_resource.nil?
+      end
+
+      def issuer_resource
+        @issuer_resource ||= resource(ISSUER_RESOURCE_NAME)
+      end
+
+      def resource(resource_name)
+        @resource_class[resource_id(resource_name)]
+      end
+
+      def resource_id(resource_name)
+        "#{account}:variable:conjur/#{authenticator_name}/#{service_id}/#{resource_name}"
+      end
+
+      def issuer_secret
+        @issuer_secret ||= @fetch_secrets.(resource_ids: [resource_id(ISSUER_RESOURCE_NAME)])
+        @issuer_secret_value ||= @issuer_secret[resource_id(ISSUER_RESOURCE_NAME)]
+      end
```

Comment:
you shouldn't have more than one memoized object in a method. the memoized object should match the name of the method. please change to this:

```suggestion
      def issuer_secret
        @issuer_secret ||= @fetch_secrets.(resource_ids: [resource_id(ISSUER_RESOURCE_NAME)])
      end
      
      def issuer_secret_value
        @issuer_secret_value ||= issuer_secret[resource_id(ISSUER_RESOURCE_NAME)]
      end
```